From 852e891955d32bbffe0780e55da5ed65148bca8a Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Thu, 13 Oct 2016 22:32:08 -0700
Subject: [PATCH 1930/2848] HADOOP-13669. Addendum patch for KMS Server should
 log exceptions before throwing.

(cherry picked from commit ae51b11f7872eaac558acf00fd23f6d7b1841cfe)
(cherry picked from commit 1ea7070042d6ed335c6c14ec3acf80e36d43b1f6)
---
 .../hadoop-kms/dev-support/findbugsExcludeFile.xml |    7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/hadoop-common-project/hadoop-kms/dev-support/findbugsExcludeFile.xml b/hadoop-common-project/hadoop-kms/dev-support/findbugsExcludeFile.xml
index bc92ed7..78c4ca6 100644
--- a/hadoop-common-project/hadoop-kms/dev-support/findbugsExcludeFile.xml
+++ b/hadoop-common-project/hadoop-kms/dev-support/findbugsExcludeFile.xml
@@ -38,4 +38,11 @@
     <Class name="org.apache.hadoop.crypto.key.kms.server.KMSWebApp"/>
     <Bug pattern="DM_EXIT"/>
   </Match>
+  <!--
+    KMS wants to log the exception before it's thrown to tomcat and disappear.
+  -->
+  <Match>
+    <Class name="org.apache.hadoop.crypto.key.kms.server.KMS"/>
+    <Bug pattern="REC"/>
+  </Match>
 </FindBugsFilter>
-- 
1.7.9.5

